internal/service/handlers/authorizeV2.go

```diff
@@ -0,0 +1,111 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"net/http"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	zkptypes "github.com/iden3/go-rapidsnark/types"
+	"github.com/rarimo/geo-auth-svc/internal/jwt"
+	"github.com/rarimo/geo-auth-svc/internal/service/requests"
+	"github.com/rarimo/geo-auth-svc/resources"
+	"gitlab.com/distributed_lab/ape"
+	"gitlab.com/distributed_lab/ape/problems"
+)
+
+func AuthorizeV2(w http.ResponseWriter, r *http.Request) {
+	req, err := requests.NewAuthorizeRequest(r)
+	if err != nil {
+		Log(r).WithError(err).Debug("failed to parse request")
+		ape.RenderErr(w, problems.BadRequest(err)...)
+		return
+	}
+
+	if !AuthVerifier(r).Disabled {
+		var proof zkptypes.ZKProof
+		if err := json.Unmarshal(req.Data.Attributes.Proof, &proof); err != nil {
+			ape.RenderErr(w, problems.BadRequest(err)...)
+			return
+		}
+
+		nullifier, err := hexutil.Decode(req.Data.ID)
+		if err != nil {
+			ape.RenderErr(w, problems.BadRequest(err)...)
+			return
+		}
```

Comment:
1) Do these checks in NewAuthorizeRequest
2) Use `format: types.ZKProof` in Swagger docs to avoid unmarshalling `req.Data.Attributes.Proof`
3) Only pass `validation.Errors` or `BadRequester interface` to `problems.BadRequest`